### PR TITLE
Pure JS runtime metrics

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
This ports the event loop metrics from C++ to JS using new APIs provided by Node. The rest of the code is copied over from its old home in a different package.

### CODE REVIEW

File: packages/solarwinds-apm/src/metrics/runtime.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    valueType: ValueType.DOUBLE,
+  }),
+  percentiles: [50, 90, 95, 99].map((p) => ({
+    p,
+    meter: meter.createObservableGauge(`eventloop.p${p}`, {
+      description: `${p}th percentile latency of the event loop`,
+      unit: "ns",
+      valueType: ValueType.DOUBLE,
+    }),
+  })),
```
Comment (raphael-theriault-swi) [inline packages/solarwinds-apm/src/metrics/runtime.ts:98]: This is new

File: packages/solarwinds-apm/src/metrics/runtime.ts [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  },
+}
+
+const elu = {
+  meter: meter.createObservableGauge("eventloop.utilization", {
+    description:
+      "percentage of time the event loop has spent processing callbacks over waiting for event",
+    unit: "%",
+    valueType: ValueType.DOUBLE,
+  }),
```
Comment (raphael-theriault-swi) [inline packages/solarwinds-apm/src/metrics/runtime.ts:131]: This is new

File: packages/solarwinds-apm/package.json [hunk 3]
```diff
@@ -92,7 +92,7 @@
     "@solarwinds-apm/eslint-config": "workspace:^",
     "@solarwinds-apm/rollup-config": "workspace:^",
     "@solarwinds-apm/test": "workspace:^",
-    "@types/node": "^20.6.0",
+    "@types/node": "^18.19.43",
```
[the remark this answers is not in the document]
Comment (raphael-theriault-swi) [reply packages/solarwinds-apm/package.json:95]: No, I just had to use that version for some weird reason I can't fully remember. Now it's gonna reflect the supported version.

**Review by tammy-baylis-swi — APPROVED** [review]
Lgtm especially the consistent usage of `description`, `unit`, and `valueType` when using Otel `meter.create*`.

**Pull request merged**